sphinx_rtd_theme/layout.html

```diff
@@ -28,16 +28,47 @@
     <link rel="canonical" href="{{ theme_canonical_url }}{{ pagename }}.html"/>
   {% endif %}
 
-  {# CSS #}
-
-  {# OPENSEARCH #}
+  {# Javascript -- Keepin in head #}
   {% if not embedded %}
-    {% if use_opensearch %}
-      <link rel="search" type="application/opensearchdescription+xml" title="{% trans docstitle=docstitle|e %}Search within {{ docstitle }}{% endtrans %}" href="{{ pathto('_static/opensearch.xml', 1) }}"/>
+    <script type="text/javascript">
+        var DOCUMENTATION_OPTIONS = {
+            URL_ROOT:'{{ url_root }}',
+            VERSION:'{{ release|e }}',
+            LANGUAGE:'{{ language }}',
+            COLLAPSE_INDEX:false,
+            FILE_SUFFIX:'{{ '' if no_search_suffix else file_suffix }}',
+            HAS_SOURCE:  {{ has_source|lower }},
+            SOURCELINK_SUFFIX: '{{ sourcelink_suffix }}'
+        };
+    </script>
+    {%- for scriptfile in script_files %}
```

Comment:
Add a comment saying not to use this? Use `app.add_javascript` instead.

Reply:
`script_files` is modified by search.html and the readthedocs-sphinx-ext, not sure what will happen if the theme removes this block.

Reply:
The extension will have to be updated. Currently `css_files` is deprecated in https://github.com/sphinx-doc/sphinx/commit/83d3fd2c365f314877086c72eef79df2c43da69d and I asked about `script_files` in https://github.com/sphinx-doc/sphinx/issues/4439

Reply:
I believe @ericholscher is planning to remove many of the `script_files` from `readthedocs-sphinx-ext`. However, who knows if others are using it.

Reply:
I think we need to _support_ people using them, since a lot of people use them. If we get to a point where Sphinx doesn't pass them anymore, then we can stop supporting them.

Reply:
There seems to be some confusing here. Note that this WILL be removed by sphinx 2.0.0 so we should add a comment in the html for people reading through the code so they know what the better option is.

Reply:
doesn't `add_javascript` just add it to `script_files` internally? So the public API is different, but for our theme it's the same API?

Reply:
Looking at the builder yes, I am still a little unsure of what direction sphinx will really go with this, so I'll ask. Since this does not affect this PR we can come back to this later.

Reply:
See this thread for why this is deprecated. So basically it can only be used for reading, not writing to unless through the sphinx app API. So we can keep `script_files` and `css_files but we should add a comment and include release notes that `extra_css_files` is deprecated so that we can avoid this type of user action. Again, we should include this in a separate commit/PR.